src/write-bin-template.ts

```diff
@@ -0,0 +1,80 @@
+import * as fs from 'fs'
+import * as path from 'path'
+import { stringify } from 'json-to-pretty-yaml'
+
+const walk = (dir: string, prevResults?: string[]): string[] => {
+  const list = fs.readdirSync(dir)
+  const results = prevResults || []
+  list.forEach((file) => {
+    file = path.resolve(dir, file)
+    const stat = fs.statSync(file)
+    if (stat.isDirectory()) {
+      walk(file, results)
+    } else {
+      results.push(file)
+    }
+  })
+  return results
+}
+
+const makeBinTemplateDirectory = (): void => {
+  const binTemplatesStat = fs.statSync('./bin/templates', {
+    throwIfNoEntry: false,
+  })
+  if (binTemplatesStat) {
+    fs.rmSync('./bin/templates', { recursive: true })
+  }
+  fs.mkdirSync('./bin/templates')
+}
+
+const getBinTargetPaths = (
+  jsonPath: string,
+): { jsonBinPath: string; yamlBinPath: string } => {
+  const jsonBinPath = jsonPath.replace('/templates/', '/bin/templates/')
+  const yamlBinPath = jsonBinPath.replaceAll('json', 'yaml')
+  return {
+    jsonBinPath,
+    yamlBinPath,
+  }
+}
+
+const makeBinTemplateSubdirectory = (jsonPath: string): void => {
+  const { jsonBinPath, yamlBinPath } = getBinTargetPaths(jsonPath)
```

Comment:
You reduce duplication here

```typescript
const makeBinTemplateDirectory(directoryPath: string): void => {
  const directoryName = path.dirname(directoryPath)
  const directoryStats = fs.statSync(directoryName, { throwIfNoEntry: false })
  if (!directoryStats?.isDirectory()) {
    fs.mkdirSync(directoryName)
  }
}

const makeBinTemplateSubdirectories = (jsonPath: string): void => {
  const { jsonBinPath, yamlBinPath } = getBinTargetPaths(jsonPath)
  [jsonBinPath, yamlBinPath].forEach((binPath) => makeBinTemplateDirectory(binPath))
}
```

Reply:
So this `makeBinTemplateSubdirectories` function is one of the things that disappeared one I started refactoring in #15. Turns out what I actually wanted was some conditions around a mkdir so I can use the same function to make the top level bin/templates and the subdirectories. 

The `forEach` has given me the idea for a func that invokes `getTemplateFileStrings` and `getBinTargetPaths` and returns an array like `{ path: string, file: string }[]`, that should have better handing in `writeFileToBin`, like

```
getWritables(jsonPath)forEach(({ path, file }) => {
      makeDirectory(path.dirname(path))
      fileSystem.writeFileSync(path, file)
    })
```